packages/flutter/lib/src/painting/cupertino_rounded_rectangle_border.dart

```diff
@@ -0,0 +1,504 @@
+        ..cubicTo(leftX(0.00000007), topY(0.68440652),
+            leftX(0.68440658), topY(-0.00000001),
+            centerX, topMidY(0))
+        ..cubicTo(centerX, topMidY(0),
+            centerX, topMidY(0),
+            centerX, topMidY(0))
+        ..lineTo(centerX, topMidY(0))
+        ..close();
+    }
+
+    Path roundedRect3b () {
+      return Path()
+        ..moveTo(centerX, topMidY(0))
+        ..lineTo(centerX, topMidY(0))
+        ..cubicTo(rightX(1.52866495), topY(0),
+            rightX(1.52866495), topY(0),
+            rightX(1.52866495), topY(0))
+        ..lineTo(rightX(1.52866495), topY(0))
+        ..cubicTo(rightX(0.68440676), topY(0.00000001),
+            rightX(0), topY(0.68440658),
+            rightMidX(0), centerY)
+        ..cubicTo(rightMidX(0), centerY,
+            rightMidX(0), centerY,
+            rightMidX(0), centerY)
+        ..cubicTo(rightMidX(0), centerY,
+            rightMidX(0), centerY,
+            rightMidX(0), centerY)
+        ..lineTo(rightMidX(0), centerY)
+        ..cubicTo(rightMidX(0), centerY,
+            rightMidX(0), centerY,
+            rightMidX(0), centerY)
+        ..lineTo(rightMidX(0), centerY)
+        ..cubicTo(rightX(0), bottomY(0.68440652),
+            rightX(0.68440676), bottomY(0),
+            rightX(1.52866495), bottomY(0))
+        ..cubicTo(rightX(1.52866495), bottomY(0),
+            rightX(1.52866495), bottomY(0),
+            rightX(1.52866495), bottomY(0))
+        ..cubicTo(rightX(1.52866495), bottomY(0),
+            rightX(1.52866495), bottomY(0),
+            rightX(1.52866495), bottomY(0))
+        ..lineTo(centerX, bottomY(0))
+        ..cubicTo(leftX(1.52866483), bottomY(0),
+            leftX(1.52866483), bottomY(0),
+            leftX(1.52866483), bottomY(0))
+        ..lineTo(leftX(1.52866471), bottomY(0))
+        ..cubicTo(leftX(0.68440646), bottomY(0),
+            leftX(-0.00000004), bottomY(0.68440676),
+            leftMidX(0), centerY)
+        ..cubicTo(leftMidX(0), centerY,
+            leftMidX(0), centerY,
+            leftMidX(0), centerY)
+        ..cubicTo(leftMidX(0), centerY,
+            leftMidX(0), centerY,
+            leftMidX(0), centerY)
+        ..lineTo(leftMidX(0), centerY)
+        ..cubicTo(leftMidX(0), centerY,
+            leftMidX(0), centerY,
+            leftMidX(0), centerY)
+        ..lineTo(leftMidX(0), centerY)
+        ..cubicTo(leftX(0.00000007), topY(0.68440652),
+            leftX(0.68440664), topY(-0.00000001),
+            leftX(1.52866483), topY(0))
+        ..cubicTo(leftX(1.52866483), topY(0),
+            leftX(1.52866483), topY(0),
+            leftX(1.52866483), topY(0))
+        ..lineTo(centerX, topMidY(0))
+        ..close();
+    }
+
+    // The radius multiplier where the resulting shape will perfectly concave at
+    // any dimension.
+    const double maxMultiplier = 3.0573;
+
+    // The multiplier used to describe the minimum radius for the dynamic shape
```

Comment:
Didn't really understand these 2 either. It's a multiplier between what and what?